iocore/hostdb/HostDB.cc

```diff
@@ -2833,15 +2650,8 @@ ParseHostFile(char const *path)
     }
   }
 
-  if (!HostFilePairs.empty()) {
-    // Need to sort by name so multiple address hosts are
-    // contiguous.
-    std::sort(HostFilePairs.begin(), HostFilePairs.end(), &CmpHostFilePair);
-    HostDBFileContinuation::scheduleUpdate(0);
-  } else if (!HostFileKeys.empty()) {
-    HostDBFileContinuation::scheduleRemove(-1, 0);
-  } else {
-    // Nothing in new data, nothing in old data, just clean up.
-    HostDBFileContinuation::finish(0);
-  }
+  // Swap out hostDB's map for ours
+  hostDB.hosts_file_ptr = parsed_hosts_file_ptr;
+  // Make a new map, so we can do it all again
+  parsed_hosts_file_ptr = new RefCountedHostsFileMap();
```

Comment:
I had the same question when I saw that, and it's likely others will wonder the same thing looking at this code in the future. It might be clearer if you do it at the start rather than at the completion of parsing.